drivers/iio/adc/adar1000.c

```diff
@@ -527,23 +527,35 @@ static int adar1000_read_adc(struct adar1000_state *st, u8 adc_ch, u32 *adc_data
 	return adar1000_mode_4wire(st, 0);
 }
 
-static void adar1000_phase_search(struct adar1000_state *st, u32 val, u32 val2,
+static void adar1000_phase_search(struct adar1000_state *st, int val, int val2,
 				  u32 *vm_gain_i, u32 *vm_gain_q, int *value_degree)
 {
 	int i, prev, next;
 
+	while (val >= 360 || val < 0) {
+		if (val >= 360)
+			val -= 360;
+		else
+			val += 360;
+	}
```

Comment:
```suggestion
val %= 360
if (val < 0)
     val = 360 - val;
```
Whouldn't be this way faster and more readable?

Reply:
That looks good to me. Shouldn't this be `val += 360`?

Reply:
yes it is val += 360 just tested it. This looks a lot nicer to me will change to it.

Reply:
my mistake, +360 :)